x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -3500,9 +3505,25 @@ export class RulesClient {
       : (rule as PartialRule<Params>);
   }
 
+  private usesValidGlobalFreqParams({
+    notifyWhen,
+    throttle,
+  }: Pick<RawRule, 'notifyWhen' | 'throttle'>) {
+    const hasNotifyWhen = typeof notifyWhen !== 'undefined';
+    const hasThrottle = typeof throttle !== 'undefined';
+    if (hasNotifyWhen && hasThrottle) return true;
+    if (!hasNotifyWhen && !hasThrottle) return false;
+    throw Boom.badRequest(
+      i18n.translate('xpack.alerting.rulesClient.usesValidGlobalFreqParams.oneUndefined', {
+        defaultMessage: 'Global notifyWhen and throttle must both be defined or both be undefined',
```

Comment:
I think the term "global" is not very self-explaining: I spent some time understanding what it means even when I was reading the tickets. Some users might think it's something global across Alerting in general, a Kibana space or Kibana as a whole. "Rule-level" might be a bit more clear terminology.